src/Application/Application.php

```diff
@@ -172,7 +172,7 @@ public function processException($e)
 			$this->httpResponse->setCode($e instanceof BadRequestException ? ($e->getCode() ?: 404) : 500);
 		}
 
-		$args = ['exception' => $e, 'request' => end($this->requests) ?: NULL];
+		$args = ['exception' => $e, 'request' => end($this->requests) ?: $e->getRequest()];
```

Comment:
This assumes that `$this->requests` can only be empty when `BadRequestException` was thrown which I'm not sure is something we can rely on.

What about rewriting the whole method like this?

~~~php
if ($e instanceof BadRequestException) {
	$httpCode = $e->getCode() ?: 404;
	$lastRequest = $e->getRequest() ?: end($this->requests);

} else {
	$httpCode = 500;
	$lastRequest = end($this->requests);

	if ($this->httpResponse instanceof Nette\Http\Response) {
		$this->httpResponse->warnOnBuffer = FALSE;
	}
}

$this->httpResponse->setCode($httpCode);
$params = ['exception' => $e, 'request' => $lastRequest];

...
~~~